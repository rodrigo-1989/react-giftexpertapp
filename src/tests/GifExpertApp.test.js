import{ shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';
import React from 'react';
describe('Evalueando componente <GifExpertApp />', () => {
    test('deberia renderizar bien los componentes', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['naruto','One punch']
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length ); 

    })
    
    
})