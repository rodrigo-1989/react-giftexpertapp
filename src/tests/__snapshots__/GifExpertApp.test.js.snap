// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evalueando componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorie={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="naruto"
      key="naruto"
    />
    <GifGrid
      category="One punch"
      key="One punch"
    />
  </ol>
</Fragment>
`;

exports[`Evalueando componente <GifExpertApp /> deberia renderizar bien los componentes 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorie={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
